calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -0,0 +1,327 @@
+        # TODO: fix Felix to support '-' characters.
+        self.pod_name = args[3].replace('-', '_')
+        self.docker_id = args[4]
+
+        print('Configuring docker container %s' % self.docker_id)
+
+        try:
+            endpoint = self._configure_interface()
+            self._configure_profile(endpoint)
+        except CalledProcessError as e:
+            print('Error code %d creating pod networking: %s\n%s' % (
+                e.returncode, e.output, e))
+            sys.exit(1)
+
+    def delete(self, args):
+        """Cleanup after a pod."""
+        self.pod_name = args[3].replace('-', '_')
+        self.docker_id = args[4]
+
+        # Remove the profile for the workload.
+        calicoctl('container', 'remove', self.docker_id)
+        calicoctl('profile', 'remove', self.pod_name)
+
+    def _configure_interface(self):
+        """Configure the Calico interface for a pod.
+
+        This involves the following steps:
+        1) Determine the IP that docker assigned to the interface inside the
+           container
+        2) Delete the docker-assigned veth pair that's attached to the docker
+           bridge
+        3) Create a new calico veth pair, using the docker-assigned IP for the
+           end in the container's namespace
+        4) Assign the node's IP to the host end of the veth pair (required for
+           compatibility with kube-proxy REDIRECT iptables rules).
+        """
+        container_ip = self._read_docker_ip()
+        self._delete_docker_interface()
+        print('Configuring Calico networking.')
+        ep = container_add(self.docker_id, container_ip, 'eth0')
+        interface_name = generate_cali_interface_name(IF_PREFIX, ep.endpoint_id)
+        node_ip = self._get_node_ip()
+        print('Adding IP %s to interface %s' % (node_ip, interface_name))
+
+        # This is slightly tricky. Since the kube-proxy sometimes
+        # programs REDIRECT iptables rules, we MUST have an IP on the host end
+        # of the caliXXX veth pairs. This is because the REDIRECT rule
+        # rewrites the destination ip/port of traffic from a pod to a service
+        # VIP. The destination port is rewriten to an arbitrary high-numbered
+        # port, and the destination IP is rewritten to one of the IPs allocated
+        # to the interface. This fails if the interface doesn't have an IP,
+        # so we allocate an IP which is already allocated to the node. We set
+        # the subnet to /32 so that the routing table is not affected;
+        # no traffic for the node_ip's subnet will use the /32 route.
+        check_call(['ip', 'addr', 'add', node_ip + '/32',
+                    'dev', interface_name])
+        print('Finished configuring network interface')
+        return ep
+
+    def _get_node_ip(self):
+        """Determine the IP for the host node.
+
+        This hits a well-known IP (the Google public DNS).
```

Comment:
Yep -- I'd like to make this better, but alas the obvious way of hitting the kube API doesn't actually work in the vagrant case. We could call 'ip addr' and parse the output I suppose.

I'm going to leave as-is for now and address later, since we need to get this in tonight.